engine/agent.py

```diff
@@ -85,4 +109,28 @@ def policy(self, decision, state_view):
 
 class BigMoneyAgent(Agent):
     def policy(self, decision, state_view):
-        pass
+        if not decision.optional and len(decision.moves) == 1:
+            return [0]
+        if decision.optional and len(decision.moves) == 0:
+            return []
+
+        if decision.prompt == 'Select a card to trash from enemy Bandit.' or \
+            decision.prompt == 'Discard down to 3 cards.':
+            return get_minimum_coin_card(decision)
+
+        if state_view.player.phase == TurnPhase.TREASURE_PHASE:
+            return [1]
+
+        if state_view.player.phase == TurnPhase.BUY_PHASE:
+            all_cards_money = [c.coins for c in state_view.player.all_cards]
```

Comment:
state_view.player gives a PlayerView object, which does not have all_cards as a member.